hvac/v1/__init__.py

```diff
@@ -526,6 +538,102 @@ def create_user_id(self, user_id, app_id, cidr_block=None, mount_point='app-id',
 
         return self._post('/v1/auth/{}/map/user-id/{}'.format(mount_point, user_id), json=params)
 
+    def create_vault_ec2_client_configuration(self, access_key, secret_key, endpoint=None):
+        """
+        POST /auth/aws-ec2/config/client
+        """
+        params = {
+            'access_key': access_key,
+            'secret_key': secret_key
+        }
+        if endpoint is not None:
+            params['endpoint'] = endpoint
+
+        return self._post('/v1/auth/aws-ec2/config/client', json=params)
+
+    def get_vault_ec2_client_configuration(self):
+        """
+        GET /auth/aws-ec2/config/client
+        """
+        return self._get('/v1/auth/aws-ec2/config/client').json()
+
+    def delete_vault_ec2_client_configuration(self):
+        """
+        DELETE /auth/aws-ec2/config/client
+        """
+        return self._delete('/v1/auth/aws-ec2/config/client')
+
+    def create_vault_ec2_certificate_configuration(self, cert_name, aws_public_cert):
+        """
+        POST /auth/aws-ec2/config/certificate/<cert_name>
+        """
+        params = {
+            'cert_name': cert_name,
+            'aws_public_cert': aws_public_cert
+        }
+        return self._post('/v1/auth/aws-ec2/config/certificate/{0}'.format(cert_name), json=params)
+
+    def get_vault_ec2_certificate_configuration(self, cert_name):
+        """
+        GET /auth/aws-ec2/config/certificate/<cert_name>
+        """
+        return self._get('/v1/auth/aws-ec2/config/certificate/{0}'.format(cert_name)).json()
+
+    def list_vault_ec2_certificate_configurations(self):
+        """
+        GET /auth/aws-ec2/config/certificates?list=true
+        """
+        params = {'list': True}
+        return self._get('/v1/auth/aws-ec2/config/certificates', params=params).json()
+
+    def create_ec2_role(self, role, bound_ami_id, role_tag=None, max_ttl=None, policies=None,
+                          disallow_reauthentication=False, **kwargs):
+        """
+        POST /auth/aws-ec2/role/<role>
+        """
+        params = {
+            'role': role,
+            'bound_ami_id': bound_ami_id,
+            'disallow_reauthentication': disallow_reauthentication
+        }
+        for param in ['role_tag', 'max_ttl', 'policies']:
+            if eval(param) is not None:
```

Comment:
I'm not a huge fan of using `eval`. Could you either make this more explicit or use `kwargs` instead?